command/agent/host_volume_endpoint.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package agent
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+func (s *HTTPServer) HostVolumesListRequest(resp http.ResponseWriter, req *http.Request) (any, error) {
+	args := structs.HostVolumeListRequest{}
+	if s.parse(resp, req, &args.Region, &args.QueryOptions) {
+		return nil, nil
+	}
+
+	query := req.URL.Query()
+	args.Prefix = query.Get("prefix")
+	args.NodePool = query.Get("node_pool")
+	args.NodeID = query.Get("node_id")
+
+	var out structs.HostVolumeListResponse
+	if err := s.agent.RPC("HostVolume.List", &args, &out); err != nil {
+		return nil, err
+	}
+
+	setMeta(resp, &out.QueryMeta)
+	return out.Volumes, nil
+}
+
+// HostVolumeSpecificRequest dispatches GET and PUT
+func (s *HTTPServer) HostVolumeSpecificRequest(resp http.ResponseWriter, req *http.Request) (any, error) {
+	// Tokenize the suffix of the path to get the volume id, tolerating a
+	// present or missing trailing slash
+	reqSuffix := strings.TrimPrefix(req.URL.Path, "/v1/volume/host/")
+	tokens := strings.FieldsFunc(reqSuffix, func(c rune) bool { return c == '/' })
+
+	switch req.Method {
+
+	// PUT /v1/volume/host
+	// POST /v1/volume/host
+	// PUT /v1/volume/host/create
+	// POST /v1/volume/host/create
+	// PUT /v1/volume/host/register
+	// POST /v1/volume/host/register
+	case http.MethodPut, http.MethodPost:
+		if len(tokens) == 0 {
+			return nil, CodedError(404, resourceNotFoundErr)
+		}
+		switch tokens[0] {
+		case "create", "":
+			return s.hostVolumeCreate(resp, req)
+		case "register":
+			return s.hostVolumeRegister(resp, req)
+		default:
+			return nil, CodedError(404, resourceNotFoundErr)
+		}
+
+	// DELETE /v1/volume/host/:id
+	case http.MethodDelete:
+		if len(tokens) == 0 {
+			return nil, CodedError(404, resourceNotFoundErr)
+		}
+		return s.hostVolumeDelete(tokens[0], resp, req)
+
+	// GET /v1/volume/host/:id
+	case http.MethodGet:
+		if len(tokens) == 0 {
+			return nil, CodedError(404, resourceNotFoundErr)
+		}
+		return s.hostVolumeGet(tokens[0], resp, req)
+	}
+
+	return nil, CodedError(404, resourceNotFoundErr)
+}
+
+func (s *HTTPServer) hostVolumeGet(id string, resp http.ResponseWriter, req *http.Request) (any, error) {
+	args := structs.HostVolumeGetRequest{
+		ID: id,
+	}
+	if s.parse(resp, req, &args.Region, &args.QueryOptions) {
+		return nil, nil
+	}
+
+	var out structs.HostVolumeGetResponse
+	if err := s.agent.RPC("HostVolume.Get", &args, &out); err != nil {
+		return nil, err
+	}
+
+	setMeta(resp, &out.QueryMeta)
+	if out.Volume == nil {
+		return nil, CodedError(404, "volume not found")
+	}
+
+	return out.Volume, nil
+}
+
+func (s *HTTPServer) hostVolumeRegister(resp http.ResponseWriter, req *http.Request) (any, error) {
+
+	args := structs.HostVolumeRegisterRequest{}
```

Comment:
Air to breathe. :grinning: 

If a function has really long signatures it'll tend to get wrapped down into the next line, and then it visually intrudes on the first line of the function body. If the first line of the function body is itself really small (like a nil check), sometimes that feels ok. But starting with a declaration and it's `{}` immediately after `() () () {` can get visually noisy.  So it's a legibility thing.

Just generally reading Go is like reading through a straw because of its lack of abstractions, so we tend to end up with _enormous_ functions and being able to add some whitespace where it feels good really helps readability.